collect_app/src/main/java/org/odk/collect/android/utilities/AuthDialogUtility.java

```diff
@@ -82,14 +84,7 @@ public void onClick(DialogInterface dialog, int which) {
                 String userNameValue = username.getText().toString();
                 String passwordValue = password.getText().toString();
 
-                // If custom username, password were passed, then we should not save the resulting credentials used
-                if (customUsername != null && customPassword != null) {
-                    webCredentialsUtils.saveCredentials(finalOverriddenUrl != null ? finalOverriddenUrl : webCredentialsUtils.getServerFromPreferences(), userNameValue, passwordValue);
-                } else if (finalOverriddenUrl == null) {
-                    webCredentialsUtils.saveCredentialsPreferences(userNameValue, passwordValue);
```

Comment:
This change here means that username and password are never saved when provided to the dialog, right? Is that intentional? Is it possible this change resulted in the issue you were seeing?

This somewhat convoluted block was added here because now it's possible to override the settings with intent extras (#2461).

Reply:
Hi @lognaturel - I moved the decision about where to save the credentials into saveCredentials - if the host is the same as what is configured in the server then save to preferences, otherwise temporarily save to memory.

Reply:
This is perhaps not super common but it could be possible for the settings to have server A configured without username and password and then for an external application to set temporary credentials to fetch or send forms. In that case, the credentials would now be saved in Collect's preferences. Did I get that right?

Reply:
Ok, i'll check that